<launch>
  <arg name="fcu_url"/>
  <arg name="auto_arm" default="false"/>
  <arg name="auto_offboard" default="false"/>
  <arg name="setpoint_subscription_topic" default="fluid/setpoint"/>
  <arg name="setpoint_publish_topic" default="mavros/setpoint_raw/local"/>
  <arg name="flip_xy" default="false"/>

 <!-- obstacle_avoidance/target_position -->
 <!-- mavros/setpoint_raw/local -->

  <rosparam command="load" file="$(find fluid)/params.yaml" />

  <param name="auto_arm" value="$(arg auto_arm)"/>
  <param name="auto_offboard" value="$(arg auto_offboard)"/>

  <node name="fluid_server" pkg="fluid" type="server" output="screen" args="$(arg setpoint_subscription_topic)"/>
  <node name="setpoint_publisher" pkg="fluid" type="setpoint_publisher" output="screen" args="$(arg setpoint_subscription_topic) $(arg setpoint_publish_topic) $(arg flip_xy)"/>

  <node name="path_optimizer" pkg="fluid" type="path_optimizer.py" output="screen"/>
  <node name="spline_generator" pkg="fluid" type="spline_generator.py" output="screen"/>
  <node name="path_publisher" pkg="fluid" type="path_publisher.py" output="screen"/>
  <node name="lqr" pkg="fluid" type="lqr.py" output="screen"/>


  <include file="$(find mavros)/launch/px4.launch">
    <arg name="fcu_url" value="$(arg fcu_url)"/>
    <arg name="log_output" value="log"/>
    <arg name="tgt_system" value="1"/>
    <arg name="tgt_component" value="1"/>
  </include>
</launch>
